### DISCUSSION TITLE
The reason for Roles?

### DISCUSSION THREAD

**Discussion by Weavor74** [TheR1D/shell_gpt#560] [Q&A]
I am guessing the role is much like a GPT from open AI, but can we use them for such detailed work. Could I create an artistic role or a python expert role. If it can be done can I call more than 1 role in a prompt, or can it call multiple roles?

**Comment by hcal** [comment]
Roles are like system prompts or "pre" prompts.  I like to pass sgpt an email and have it rewrite it for clarity.  So I have a role set up with the instructions "Write a refined and professional email based on the provided draft. Avoid introductory phrases such as "I hope this email finds you well." Maintain a tone and style similar to contemporary business literature. Use common American phrasing and vocabulary, and steer clear of non-American English expressions or styles."

By saving it as a role, I dont have to type it over everytime.
